service/awsconfig/v7/adapter/lifecycle_hooks.go

```diff
@@ -0,0 +1,32 @@
+package adapter
+
+import "github.com/giantswarm/aws-operator/service/awsconfig/v7/key"
+
+// The template related to this adapter can be found in the following import.
+//
+//     github.com/giantswarm/aws-operator/service/awsconfig/v7/templates/cloudformation/guest/lifecycle_hooks.go
+//
+
+type lifecycleHooksAdapter struct {
```

Comment:
In order to inspect lifecycle states of instances we need to add lifecycle hooks. Therefore the adapter.